compiler/rustc_error_codes/src/lib.rs

```diff
@@ -397,7 +397,6 @@ E0618: 0618,
 E0619: 0619,
 E0620: 0620,
 E0621: 0621,
-E0622: 0622,
```

Comment:
Please don't remove it but instead add a comment after with `// REMOVED: explanation`

Reply:
Dealing with no-longer emitted error codes is a pain. I added some comments at the top of this file to explain what one has to do (which I had to figure out by trial and error as nothing seemed documented). Not sure how to make this more visible.

Reply:
Agreed but `--explain` is still supposed to work. ^^'

Reply:
Wouldn't it make sense to keep this error for misused `#[rustc_intrinsic]`?
Currently that annotation appears to be silently ignored (or ICE) when misapplied [playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=a4f33b34985e9a5f73346f9bd41e76b5)

For this reason I hadn't removed the check motioned in https://github.com/rust-lang/rust/pull/139530#discussion_r2033306499

Reply:
Currently, `rustc_*` attributes are excluded from the "must have logic to check whether they are used in the correct place" rule. Some have impls regardless, but most do not.